operator/api/v1beta1/alertingrule_types.go

```diff
@@ -0,0 +1,134 @@
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// AlertingRuleSpec defines the desired state of AlertingRule
+type AlertingRuleSpec struct {
+	// Tenant to associate the alerting rule groups.
+	//
+	// +required
+	// +kubebuilder:validation:Required
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Tenant ID"
+	TenantID string `json:"tenantID"`
+
+	// List of groups for alerting rules.
+	//
+	// +optional
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Groups"
+	Groups []*AlertingRuleGroup `json:"groups"`
+}
+
+// AlertingRuleGroup defines a group of Loki alerting rules.
+type AlertingRuleGroup struct {
```

Comment:
What is the benefit of such a struct? IMHO it makes writing the golang types a bit harder, because one needs to type the extra embedded types for the common fields.

Reply:
Can you draft an alternative in a couple of lines of code here? I personally tolerate the duplication here, because APIs and their code should move version independently, e.g. one gets promoted to v1 before the other.

Reply:
Per our online discussion, this will be resolved for the `internal/manifests` code path in a separate issue that addresses configmap sharding for alerting/recording rules.